### PR TITLE
feat: add update backup target action

### PR DESCRIPTION
**Pull request by a110605**
### What this PR does / why we need it
 - [x] Add volume update backup target action
 - [x] Add volume bulk update backup target action
 - [x] Fix  `Delete All backups` action in backup detail page
 - [x] Remove unneeded `volumeName` feild in backup detail page url
    - e.g. http://localhost:8080/#/backup/vol2?field=name&keyword=vol2-ac9ede2e

### Issue
https://github.com/longhorn/longhorn/issues/8647

### Test Result

```
LONGHORN_MANAGER_IP=http://152.42.233.250:30001/ npm run dev
```

**Update volume backup target** 

https://github.com/user-attachments/assets/6e1bb915-cb17-4728-8dcb-744749e715cc

**Bulk update backup target** 

https://github.com/user-attachments/assets/17260f15-02ff-4746-a804-15c0e6bdca3d

Fix action in backup detail page

Screenshot 2024-12-19 at 10 49 48 AM

## Release Notes

- **New Features**
  - Introduced modals for updating backup targets, enhancing user interaction for both single and bulk updates.
  - Improved backup filtering and error handling for better user feedback.
  - Added functionality for managing backup target updates in the volume management workflow.

- **Bug Fixes**
  - Enhanced the logic for identifying current backups based on query parameters.

- **Refactor**
  - Streamlined method names for consistency and clarity.
  - Updated state management for volume operations to improve user experience.

- **Chores**
  - Updated prop types and method signatures to reflect new functionalities.

### CODE REVIEW

File: src/routes/backup/BackupDetail.js [hunk 1]
```diff
@@ -22,15 +22,15 @@ function Backup({ backup, volume, setting, backingImage, loading, location, disp
...
+  const { keyword } = queryString.parse(location.search)
+  const currentBackUp = backupVolumes.find(bkVol => bkVol.name === keyword) || {}
   const v1DataEngineEnabledSetting = settings.find(s => s.id === 'v1-data-engine')
   const v2DataEngineEnabledSetting = settings.find(s => s.id === 'v2-data-engine')
   const v1DataEngineEnabled = v1DataEngineEnabledSetting?.value === 'true'
   const v2DataEngineEnabled = v2DataEngineEnabledSetting?.value === 'true'
 
-  const backups = volumeName ? backupData.filter(bk => bk.volumeName === volumeName) : backupData
-  sortBackups(backupData)
+  const backups = currentBackUp ? backupData.filter(bk => bk.volumeName === currentBackUp.volumeName) : backupData
```
Comment (houhoucoop) [inline src/routes/backup/BackupDetail.js:32]: We might need to add the `backuptarget` filter to the URL as well?
The scenarios are as follows:
- Create a volume with the backup target `default` and create a backup. (e.g. 1217-test)
- Change the backup target to `cifs` and create another backup.
- On the backup page, both backups appear with their respective targets (expected).
Screenshot 2024-12-20 at 10 40 38 AM (2)
- Clicking the volume with the `default` target shows `cifs` in the details page.
Screenshot 2024-12-20 at 10 43 11 AM (2)

Comment (houhoucoop) [reply src/routes/backup/BackupDetail.js:32]: Another issue found: When clicking `Restore Latest Backup` for 1217-test (backup target: default), the modal doesn’t appear.

File: src/models/backup.js [hunk 2]
```diff
@@ -363,14 +363,13 @@ export default {
...
+      const targetBackupVolume = state.backupVolumes.find((item) => item.name === volumeId) || {}
+      if (targetBackupVolume?.volumeName && action.payload?.data) {
+        const backupData = action.payload.data.filter((item) => item.volumeName === targetBackupVolume?.volumeName)
         return {
           ...state,
-          backupData: backupData || [],
+          backupData,
         }
       }
       return {
```
Comment (houhoucoop) [inline src/models/backup.js:375]: Found a issue in backing image page, when accessing `/backingImage` directly, the `Back up` button is disabled.
ref: https://github.com/longhorn/longhorn-ui/pull/755/files#diff-bbfb23ee1c5b3de51151593ac9fe60b487342f1265e1195ac59497efe9f9ed9cR188

Comment (a110605) [reply src/models/backup.js:375]: Fixed in [d68cc66](https://github.com/longhorn/longhorn-ui/pull/830/commits/d68cc666bcf27feaf2bc3d0e7383410a8fc5de14)

File: src/routes/backup/BackupDetail.js (hunk 1 above)
Comment (a110605) [reply src/routes/backup/BackupDetail.js:32]: Create Diasater Recovery Volime for 1217-test (default) is good for me.
image

Comment (a110605) [reply src/routes/backup/BackupDetail.js:32]: Like the backend bug that the `/v1/backupvolumes/bug-test-a98258ed?action=backupList` (default) return cifs backup data. @mantissahz will fix from backend first.

Comment (a110605) [reply src/routes/backup/BackupDetail.js:32]: @houhoucoop , @mantissahz , let's review and merge this PR to add update backup target feature first. We can fix this bug in next PR working with backend change.

**Review by houhoucoop — APPROVED** [review]
LGTM

**Comment by a110605** [comment]
[@mergify](https://github.com/mergify) backport v1.8.x

**Pull request merged**